Neos.Flow/Tests/Unit/ObjectManagement/Proxy/ProxyClassTest.php

```diff
@@ -52,7 +66,16 @@ public function proxyClassesDataProvider()
                 'expectedProxyCode' =>
                     'class ClassWithoutNamespace extends ClassWithoutNamespace' . Compiler::ORIGINAL_CLASSNAME_SUFFIX . " implements \\Neos\\Flow\\ObjectManagement\\Proxy\\ProxyInterface {\n\n" .
                     "    const TEST_CONSTANT = 1;\n\n" .
-                    '}',
+                    "}\n\n" .
+                    "class ClassWithoutNamespace_LazyProxy extends ClassWithoutNamespace implements \Neos\Flow\ObjectManagement\DependencyInjection\DependencyProxy\n" .
+                    "{\n" .
+                    "    use \Neos\Flow\ObjectManagement\DependencyInjection\DependencyProxyTrait;\n\n" .
+                    "    public function __call(string \$methodName, array \$arguments)\n" .
+                    "    {\n" .
+                    "        [\$methodName, \$arguments] = func_get_args();\n" .
```

Comment:
why do we need this line? arent those parameter declared as method argument?

i i guess, if someone writes a `__call` method on their own class so its compatible?

Reply:
Yes, just to make sure that the method signature is compatible.